Src/SampQuery.cs

```diff
@@ -382,15 +444,15 @@ private ServerRules CollectServerRulesFromByteArray(byte[] data) {
 
                     for (int i = 0, iRules = read.ReadInt16(); i < iRules; i++)
                     {
-                        PropertyInfo property = sampServerRulesData.GetType().GetProperty(new string(read.ReadChars(read.ReadByte())).Replace(' ', '_'), BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
+                        PropertyInfo? property = sampServerRulesData.GetType().GetProperty(new string(read.ReadChars(read.ReadByte())).Replace(' ', '_'), BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
                         value = new string(read.ReadChars(read.ReadByte()));
 
                         if (property != null)
                         {
                             if (property.PropertyType == typeof(bool)) val = value == "On";
                             else if (property.PropertyType == typeof(Uri)) val = Helpers.ParseWebUrl(value);
                             else if (property.PropertyType == typeof(DateTime)) val = Helpers.ParseTime(value);
-                            else val = Convert.ChangeType(value, property.PropertyType, CultureInfo.InvariantCulture);
+                            else val = Helpers.TryParseByte(value, property);

```

Comment:
This block should be refactored, properties need to be checked by name and not by type. What about TryParseByte, it should be tested and unit tests should be written.

Reply:
Also we need the unit tests for GetServerIsOMP as well